py3status/modules/uptime.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+"""
+Display system uptime.
+
+Configuration parameters:
+    default_format: start uptime in this format: 'pretty', 'short', or 'tiny'
+        (default 'pretty')
+    format: display format for this module (default 'up {format_uptime}')
+    format_pretty: show uptime in pretty format
+        *(default '[\?if=W {W} weeks, ][\?if=D {D} days, ]
+        [\?if=H {H} hours, ][\?if=M {M} minutes, ]')*
+    format_short: show uptime in short format
+        (default '[\?if=W {W}w ][\?if=D {D}d ][\?if=H {H}h ][\?if=M {M}m ]')
+    format_tiny: show uptime in tiny format
+        (default '[\?if=D {D}, ][\?if=H {H}:][\?if=M {M}:]')
+
+Datetime placeholders:
+    {G} Number of decades
+    {Y} Number of years
+    {W} Number of weeks
+    {D} Number of days
+    {H} Number of hours
+    {M} Number of minutes
+    {S} Number of seconds
+
+Format placeholders:
+    {format_uptime} uptime in multiple formats
+        To remove time clicking fun, remove {format_uptime}
+        and update format with desired datetime placeholders above.
+
+Format_* placeholders:
+    See datetime placeholders above.
+
+Note: If you don't use one of the placeholders, the value will be carried over
+to the next unit! For example, given an uptime of 1h30min:
+    If you use {minutes} as your only placeholder, then its value will be 90.
+    If you use {hours} and {minutes}, then its values will be 1 and 30, respectively.
+
+@author Alexis "Horgix" Chotard <alexis.horgix.chotard@gmail.com>
+@license BSD
+
+SAMPLE OUTPUT
+{'full_text': 'up 1 days, 18 hours, 20 minutes'}
+
+short
+{'full_text': 'up 1d 18h 20m'}
+
+tiny
+{'full_text': 'up 1, 18:20'}
+"""
+
+from time import time
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    default_format = 'pretty'
+    format = 'up {format_uptime}'
+    format_pretty = '[\?if=W {W} weeks, ][\?if=D {D} days, ]' + \
+        '[\?if=H {H} hours, ][\?if=M {M} minutes, ]'
+    format_short = '[\?if=W {W}w ][\?if=D {D}d ][\?if=H {H}h ][\?if=M {M}m ]'
+    format_tiny = '[\?if=D {D}, ][\?if=H {H}:][\?if=M {M}:]'
```

Comment:
>adding multiple formats

Module enhancement. Uptime is so boring if it can't do anything else. This now display uptime in few different formats. I skipped `since` format because it's too different, but will leave it to others if they want to include this format. I don't.

>obscure placeholder names is not the way to go imho.

I agree, but this is timedate. I feel we should make an exception for this. See `pomodoro` for an example. Pomodoro uses `{ss}`, `{mm}`, etc. I went with simple variables similar to `man date` for placeholders. All of them except for `{G}` to represent decades.

If we spell it out, then the format gets awful too long, then I made multiple `format_minutes` things... then we say we want everything in one format instead, but with full datetime names in one format... along with the formatter... it gets awfully too long.

If I go back to `format_minutes` thing, then this will be similar to upcoming `weather_owm` module where they allows custom formats for each `{weather}` thing.

I think, in a nutshell, that it's best we make an exception for timedate so we can put everything in one format (including the formatter) like what I did here. The multiple formats are the module enhancement that I'd like to add.